pkg/solana/chainwriter/chain_writer.go

```diff
@@ -0,0 +1,324 @@
+			filteredLookupTables[tableKey] = usedAddresses
+		}
+	}
+
+	return filteredLookupTables
+}
+
+func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contractName, method string, args any, transactionID string, toAddress string, meta *types.TxMeta, value *big.Int) error {
+	programConfig, exists := s.config.Programs[contractName]
+	if !exists {
+		return fmt.Errorf("failed to find program config for contract name: %s", contractName)
+	}
+	methodConfig, exists := programConfig.Methods[method]
+	if !exists {
+		return fmt.Errorf("failed to find method config for method: %s", method)
+	}
+
+	// Configure debug ID
+	debugID := ""
+	if methodConfig.DebugIDLocation != "" {
+		var err error
+		debugID, err = GetDebugIDAtLocation(args, methodConfig.DebugIDLocation)
+		if err != nil {
+			return errorWithDebugID(fmt.Errorf("error getting debug ID from input args: %w", err), debugID)
+		}
+	}
+
+	codec := s.codecs[contractName]
+	encodedPayload, err := codec.Encode(ctx, args, method)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error encoding transaction payload: %w", err), debugID)
+	}
+
+	// Fetch derived and static table maps
+	derivedTableMap, staticTableMap, err := s.ResolveLookupTables(ctx, args, methodConfig.LookupTables)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error getting lookup tables: %w", err), debugID)
+	}
+
+	// Resolve account metas
+	accounts, err := GetAddresses(ctx, args, methodConfig.Accounts, derivedTableMap, s.reader)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error resolving account addresses: %w", err), debugID)
+	}
+
+	// Filter the lookup table addresses based on which accounts are actually used
+	filteredLookupTableMap := s.FilterLookupTableAddresses(accounts, derivedTableMap, staticTableMap)
+
+	// Fetch latest blockhash
+	blockhash, err := s.reader.LatestBlockhash(ctx)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error fetching latest blockhash: %w", err), debugID)
+	}
+
+	// Prepare transaction
+	programID, err := solana.PublicKeyFromBase58(toAddress)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error parsing program ID: %w", err), debugID)
+	}
+
+	feePayer, err := solana.PublicKeyFromBase58(methodConfig.FromAddress)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error parsing fee payer address: %w", err), debugID)
+	}
+
+	tx, err := solana.NewTransaction(
+		[]solana.Instruction{
+			solana.NewInstruction(programID, accounts, encodedPayload),
+		},
+		blockhash.Value.Blockhash,
+		solana.TransactionPayer(feePayer),
+		solana.TransactionAddressTables(filteredLookupTableMap),
+	)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error constructing transaction: %w", err), debugID)
+	}
+
+	// Enqueue transaction
+	if err = s.txm.Enqueue(ctx, accounts[0].PublicKey.String(), tx, &transactionID); err != nil {
+		return errorWithDebugID(fmt.Errorf("error enqueuing transaction: %w", err), debugID)
+	}
```

Comment:
Enqueue now needs lastValidBlockHeight param too. You can access it through blockhash.Value.LastValidBlockHeight